### ISSUE TITLE
[FEA] Add to_host method to column_wrapper test classes

### ISSUE THREAD

**Issue by davidwendt** [rapidsai/cudf#3216]
The legacy gtests class `column_wrapper` had a `to_host()` and an `operator==()` methods to help verify and diagnose test results.

The `expect_columns_equal()` utility should be used in place of `operator==` but `to_host()` would still be useful.

Proposal from a slack discussion is for `fixed_width_column_wrapper::to_host` return a `pair<vector<T>,vector<bitmask_type>>` and the `strings_column_wrapper::to_host` return `pair<std::vector<string>, vector<bitmask_type>>`

@jrhemstad @devavret

**Comment by jrhemstad** [comment]
I thought about this more, and these should be free functions that act on `column_view` and `strings_column_view` objects respectively.

E.g.,

```
// explicit template param requires caller to specify the type
// should perform a check to make sure that `type_to_id<T>() == c.type()`
template <typename T>
std::pair<std::vector<T>, std::vector<bitmask_type>> to_host(column_view c);

std::pair<std::vector<std::string>, std::vector<bitmask_type>> to_host(strings_column_view s);
```

This way we can move columns to the host that may not necessarily be wrapped in a `column_wrapper` object.

**Comment by devavret** [comment]
@jrhemstad What's the use case of host copy outside of testing?

**Comment by jrhemstad** [comment]
> @jrhemstad What's the use case of host copy outside of testing?

It's still for testing. My suggestion about making it a free function will allow using `to_host()` on a column returned from a libcudf function (i.e., one that isn't wrapped in a `column_wrapper`).

**Comment by devavret** [comment]
The free function `to_host` needs two changes over the `column_wrapper` member:
1. The `column_view_type` would need to be a template because it should work on both `column_view` and `mutable_column_view`.
2. The copy would be slightly trickier for null mask as we'd expect the mask to be copied from the offset onwards.

**Comment by jrhemstad** [comment]
> 1. The `column_view_type` would need to be a template because it should work on both `column_view` and `mutable_column_view`.

`mutable_column_view` is implicitly convertible to `column_view`, so there's no need to template it.

> The copy would be slightly trickier for null mask as we'd expect the mask to be copied from the offset onwards.
[This](https://github.com/rapidsai/cudf/pull/3219/files#diff-db436947957f64f102c5ddb967a1374bR198) should help with that https://github.com/rapidsai/cudf/pull/3219

**Comment by jrhemstad** [comment]
@devavret can you break out `to_host` as a separate PR to unblock https://github.com/rapidsai/cudf/pull/3255?

**Comment by devavret** [comment]
`to_host` is still dependent on #3219

**Comment by jrhemstad** [comment]
> `to_host` is still dependent on #3219

Yep, working with @kaatish to get that done.

**Comment by jrhemstad** [comment]
@devavret are you also planning to do the string version? Or would you like @davidwendt to assist with that?

**Comment by devavret** [comment]
Not the string version no. I don't have the bandwidth currently.

**Comment by jrhemstad** [comment]
@devavret can you open a PR for your `to_host` implementation?

**Comment by devavret** [comment]
There's a mess of dependencies in my PR now. Can you just merge #3237 ?

**Comment by jrhemstad** [comment]
> There's a mess of dependencies in my PR now. Can you just merge #3237 ?

Looks like #3237 only depends on https://github.com/rapidsai/cudf/pull/3219, and that's only for `to_host`. #3219 is not ready to merge. 

`to_host` is blocking https://github.com/rapidsai/cudf/pull/3255 and having a separate PR will allow @davidwendt to contribute a version for strings without having to push to #3237.
